internal/httpserve/handlers/impersonation.go

```diff
@@ -0,0 +1,327 @@
+package handlers
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/rs/zerolog/log"
+	echo "github.com/theopenlane/echox"
+	"github.com/theopenlane/iam/auth"
+	"github.com/theopenlane/iam/tokens"
+
+	"github.com/theopenlane/core/internal/ent/generated"
+	"github.com/theopenlane/core/pkg/models"
+	"github.com/theopenlane/utils/rout"
+)
+
+// Error definitions for impersonation operations
+var (
+	// ErrAuthenticationRequired indicates that the user must be authenticated to perform this action
+	ErrAuthenticationRequired = errors.New("authentication required")
+	// ErrNoActiveImpersonationSession indicates that there is no active impersonation session
+	ErrNoActiveImpersonationSession = errors.New("no active impersonation session")
+	// ErrInvalidSessionID indicates that the provided session ID is invalid
+	ErrInvalidSessionID = errors.New("invalid session ID")
+	// ErrInsufficientPermissionsSupport indicates that the user does not have permissions to perform support impersonation
+	ErrInsufficientPermissionsSupport = errors.New("insufficient permissions for support impersonation")
+	// ErrInsufficientPermissionsAdmin indicates that the user does not have permissions to perform admin impersonation
+	ErrInsufficientPermissionsAdmin = errors.New("insufficient permissions for admin impersonation")
+	// ErrJobImpersonationAdminOnly indicates that job impersonation is only allowed for system admins
+	ErrJobImpersonationAdminOnly = errors.New("job impersonation only allowed for system admins")
+	// ErrInvalidImpersonationType indicates that the provided impersonation type is invalid
+	ErrInvalidImpersonationType = errors.New("invalid impersonation type")
+	// ErrTargetUserNotFound indicates that the target user for impersonation was not found
+	ErrTargetUserNotFound = errors.New("target user not found")
+	// ErrTokenManagerNotConfigured indicates that the token manager is not configured
+	ErrTokenManagerNotConfigured = errors.New("token manager not configured")
+	// ErrFailedToExtractSessionID indicates that the session ID could not be extracted from the token
+	ErrFailedToExtractSessionID = errors.New("failed to extract session ID from token")
+)
+
+// StartImpersonation handles requests to start user impersonation
+func (h *Handler) StartImpersonation(ctx echo.Context) error {
+	var req models.StartImpersonationRequest
+	if err := ctx.Bind(&req); err != nil {
+		return h.InvalidInput(ctx, err)
+	}
+
+	if err := req.Validate(); err != nil {
+		return h.InvalidInput(ctx, err)
+	}
+
+	// Get the current authenticated user (the impersonator)
+	currentUser, ok := auth.AuthenticatedUserFromContext(ctx.Request().Context())
+	if !ok {
+		return h.Unauthorized(ctx, ErrAuthenticationRequired)
+	}
```

Comment:
I would recommend using `auth.GetAuthenticatedUserFromContext(ctx)` which checks both the ok + ensures current user is not nil

```suggestion
	// Get the current authenticated user (the impersonator)
	currentUser, err := auth.GetAuthenticatedUserFromContext(ctx.Request().Context())
	if err != nil {
		return h.Unauthorized(ctx, ErrAuthenticationRequired)
	}
```